adoc/extensions/sycl_khr_group_interface.adoc

```diff
@@ -0,0 +1,849 @@
+by this extension all implement a common set of operations, which are shown in
+the synopsis below.
+The name [code]#+__group__+# in that synopsis is a placeholder for each of these
+three types.
+When the synopsis shows an ellipsis ([code]#+/*...*/+#), the subsequent sections
+clarify the definition for each type.
+
+Whether it is possible to query certain properties of a group at compile-time
+depends on the group's type.
+A member function that is not declared [code]#constexpr# in the synopsis below
+may still be declared [code]#constexpr# for a specific group type defined in a
+subsequent section.
+
+[source,role=synopsis]
+----
+
+namespace sycl::khr {
+
+class __group__ {
+
+ public:
+  using id_type = /* ... */;
+  using linear_id_type = /* ... */;
+  using range_type = /* ... */;
+  using extents_type = /* ... */; // C++23
+  using size_type = /* ... */;
+  static constexpr int dimensions = /* ... */;
+  static constexpr memory_scope fence_scope = /* ... */;
+
+  id_type id() const noexcept;
+  linear_id_type linear_id() const noexcept;
+
+  range_type range() const noexcept;
+
+  extents_type extents() const noexcept; // C++23
+  extents_type::index_type extent(extents_type::rank_type r) const noexcept; // C++23
+
+  constexpr static extents_type::rank_type rank() noexcept; // C++23
+  constexpr static extents_type::rank_type rank_dynamic() noexcept; // C++23
+  constexpr static size_t static_extent(rank_type r) noexcept; // C++23
+
+  size_type size() const noexcept;
+
+};
+
+template <typename Group>
+member_item<Group> get_item(Group g) noexcept;
+
+template <typename Group>
+bool leader_of(Group g) noexcept;
+
+} // namespace sycl::khr
+----
+
+[[sec:khr-group-interface-common-member-funcs]]
+=== Member functions
+
+.[apidef]#+__group__::id+#
+[source,role=synopsis,id=api:khr-group-interface-common-group-id]
+----
+id_type id() const noexcept;
+----
+
+_Returns_: The index of this group within the index space returned by
+[api]#+__group__::range+#.
+
+'''
+
+.[apidef]#+__group__::linear_id+#
+[source,role=synopsis,id=api:khr-group-interface-common-group-linear-id]
+----
+linear_id_type linear_id() const noexcept;
+----
+
+_Returns_: The linearized index (see <<sec:multi-dim-linearization>>) of this
+group within the index space returned by [api]#+__group__::range+#.
+
+'''
+
+.[apidef]#+__group__::range+#
+[source,role=synopsis,id=api:khr-group-interface-common-group-range]
+----
+range_type range() const noexcept;
+----
+
+_Returns_: An index space representing the collection of groups that includes
+this group, and which defines the range of valid [code]#id# values for this
+group.
+
+'''
+
+.[apidef]#+__group__::extents+#
+[source,role=synopsis,id=api:khr-group-interface-common-group-extents]
+----
+constexpr extents_type extents() const noexcept;
+----
+
+_Minimum C++ Version_: {cpp23}
+
+_Returns_: The number of work-items in each dimension of the group.
```

Comment:
Am I correct in thinking that the value returned by `extents()` equivalent to `range()` but as an extents type?

Reply:
No: `extents()` is the (multi-dimensional) size of the group, whereas `range()` is the number of groups.

The way to remember this (and also what was discussed in [this comment](https://github.com/KhronosGroup/SYCL-Docs/pull/638#discussion_r1973906024)) is that the goal of this redesign is to clarify which properties belong to the group.  So:
- `id` is the index _of the group_
- `range` is the range _of the group's indices_ (like how `item.get_range()` returns the range of an item's indices)
- `extents` is the extents _of the group_
- `size` is the size _of the group_

Reply:
Exactly.